t/macros/load_macros.t

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env perl
+
+=head1 Test macro loading
+
+This tests the pg/macro files to see that they all load without errors or
+warnings.
+
+Note that macros that are broken are listed in C<%brokenMacros> and testing of
+these macros is skipped.  These macros should be fixed or considered for
+deletion.
+
+=cut
+
+die "PG_ROOT not found in environment.\n" unless $ENV{PG_ROOT};
+use lib "$ENV{PG_ROOT}/lib";
+
+use Test2::V0;
+use File::Find;
+use WeBWorK::PG;
+
+my %baseMacros = ('PG.pl' => 1, 'PGstandard.pl' => 1);
+
+my %brokenMacros = ('answerDiscussion.pl' => 1);
+
+die "Unable to open pg macro directory: $ENV{PG_ROOT}/macros" unless -e "$ENV{PG_ROOT}/macros";
+
+# Find all macro files inside the $ENV{PG_ROOT}/macros directory.
+my @macro_files;
+find ( sub {
+		my $f = $_;
+		return unless -f;       #Must be a file
+		return unless /\.pl$/;  #Must end with `.pl` suffix
+		return if grep { $f eq $_ } (keys(%baseMacros),keys(%brokenMacros));
+		push @macro_files, $File::Find::name;
```

Comment:
Many of the macro files are failing the test that were not failing before.  The reason for this is that macros are loaded by file name only, not with full path.  This needs to be
```suggestion
		push @macro_files, $File::Find::name =~ s!.*/!!r;
```
so that the path is removed and the macros are loaded correctly.

Reply:
This is still incorrect.  Remove the dependence on `File::Basename` and change this as I said above.  Just use `$_` which is the basename of the file to begin with.  Remember to add `$pg_root/macros/capa` to the `macrosPath` array in pg_config.dist.yml.

Reply:
Missed this comment above.  Will update.

Reply:
The point here is that on what is currently line 46 the basename of the file must be used.  With the current code the full path of the file is used because that is what you are putting into the @macro_files array.